templates/profile/hathitrust/solr6/index-release.sh.erb

```diff
@@ -1,117 +1,85 @@
-#! /bin/bash
+#!/bin/bash
 
-TODAY=`date +%Y-%m-%d`
-#TODAY="2019-08-31"
+onoe () {
+  echo "${1}"
+  exit 1
+}
 
-BASE=/htsolr/<%= @solr_name %>
-SYMLINKBASE=/htsolr/serve
-SNAP=".snapshot/<%= @snapshot_name %>_${TODAY}"
-CORES="<%= @solr_cores.join(' ') %>"
+base=/htsolr/<%= @solr_name %>
+serve=/htsolr/serve
+snaps=( "${base}/.zfs/snapshot/htsolr-<%= @solr_name %>_"* )
+snap="${snaps[-1]}"
+snapname="$(basename "${snap}")"
 
 # check if the administrative release stop flag is set
-if [ -f ${BASE}/flags/<%= @solr_stop_flag %> ]; then
-  /bin/echo "<%= @solr_stop_flag %> flag present...skip release today"
-  exit 1
-fi
+[[ -f "${base}/flags/<%= @solr_stop_flag %>" ]] && onoe "<%= @solr_stop_flag %> flag present...skip release today"
 
 # check that today's snap exists
-for s in $CORES; do
-  if [ ! -d $BASE/cores/${s}/${SNAP} ]; then
-    /bin/echo "missing today's snapshot...aborting release"
-    exit 1
-  fi
-done
+[[ ! -d ${snap}/cores/ ]] && onoe "can't read snapshot dir...aborting release"
+[[ ! -d ${snap}/flags/ ]] && onoe "can't read snapshot dir...aborting release"
 
 # check today's snap for 'busy' flag
-if [ -f $BASE/flags/${SNAP}/busy ]; then
-  /bin/echo "indexing and/or optimization ocurring...skip release today"
-  exit 1
-fi
+[[ -f "${snap}/flags/busy" ]] && onoe "indexing or optimization ocurring...skip release today"
 
 <% if @is_primary_site %>
-# # check whether release happened at mirror site
-# if ! curl -A SOLR -s --retry 5 --fail https://babel.hathitrust.org/flags/web/<%= @solr_name %>-release-${TODAY} --resolve "babel.hathitrust.org:443:<%= @mirror_site_ip %>" >& /dev/null; then
-#   /bin/echo "Mirror site index release appears to have failed...skip release today"
-#   exit 1
-# fi
-<% end %>
+# TODO: check whether release happened at mirror site
+# ...
 
-for s in $CORES; do
-  SEGMENTS=`ls ${BASE}/cores/${s}/${SNAP}/<%= @core_data_dir_template %>/index/*.fdt | wc -l`
-  <% if @is_catalog %>if [ ${SEGMENTS} -eq 0 ]; then<% end %>
-  <% if @is_lss     %>if [ ${SEGMENTS} -lt 1 ] || [ ${SEGMENTS} -gt 2 ]; then<% end %>
-    /bin/echo "at least one core served by `hostname -s` has an improper number of segments...aborting release"
-    exit 1
-  fi
-done
+<% end %>
+# check that cores appear valid
+<% @solr_cores.each do |core| %>
+[[ ! -d ${snap}/cores/<%= core %>/data/index ]] && onoe "core data missing...aborting release"
+segments=( "${snap}/cores/<%= core %>/data/index/"*.fdt )
+(( ${#segments[@]} > 0 )) || onoe "core '<%= core %>' served by $(hostname -s) has an improper number of segments...aborting release"
```

Comment:
This appears to check that the core has at least 1 segment. That loses the check that full-text has either 1 or 2 segments. While for full text such an index might or might not work, it's probably an indication that the optimization process didn't complete, and we probably wouldn't want to do a release. Normally, the busy flag should be set in such a case, but I could imagine a case where something went wrong, the busy flag isn't set, optimize didn't finish, and the core is in an inconsistent state. That might be unlikely enough (> 2 segments, busy flag not set, index not usable) it isn't worth checking for here, though.